CLIENT_SPECIFICATION.md

```diff
@@ -0,0 +1,206 @@
+ - v2.0 (12th January 2019)
+	 - Expanded notes into full sentences
+
+
+## Terminology
+In order to aid the understanding of this specification document, a number of terms will be defined in this section.
+
+### Page
+tldr-pages consists of multiple _pages_ - each of which describes a specific command.
+
+### Platform
+Pages are grouped by platform. A platform is an operating system. For example, `windows`, `linux`, `osx`. The special platform `common` contains pages for commands that work identically across all platforms.
+
+
+## Command-Line Interface
+This section describes the standardised command-line interface (CLI) for clients implementing one. Clients that do not provide a CLI can ignore this section.
+
+### Arguments
+A number of arguments MUST be supported (unless otherwise specified if a CLI is implemented:
+
+Argument		        | Meaning
+------------------------|--------------------
+`--update`				| Updates the offline cache of pages. MUST be implemented if cache is supported.
+`--version`, `-v`		| Shows the current version of the client, and the version of this specification that it implements.
+`--list`, `-l`			| Lists all the pages in the current platform to the standard output. If the special platform `all` is specified a list of all pages in all platforms MUST be displayed.
+`--pretty`				| OPTIONAL. Forces the output to contain all additional decorations, even if the standard output is not a TTY.
+`--platform`, `-p`		| Specifies the platform that should be used for resolving page names. If specified, the named platform MUST be checked first instead of the host platform  as described below.
+
+Additional decoration MAY be printed if the standard output is a [TTY](http://www.linusakesson.net/programming/tty/index.php). If not, then the output MUST not contain any additional decorations. For example a page list MUST be formatted with 1 page name per line (to enable easy manipulation using standard CLI tools such as `grep` etc.).
+
+Clients MAY support additional custom arguments and syntax not documented here.
+
+Here are some examples invocations using the above flags:
+
+```bash
+tldr --update
+tldr --version
+tldr -l
+```
+
+### Page Names
+The first argument that does not start with a dash (`-`) MUST be considered the page name.
+
+In addition, page names MAY contain spaces (e.g. `git status`) - such page names MUST be transparently concatenated with dashes (`-`). For example, the page name:
+
+```
+git checkout
+```
+
+...becomes this:
+
+```
+git-checkout
+```
+
+Here are some example invocations:
+
+```bash
+tldr 7za
+tldr eyeD3
+tldr git checkout
+# In the below, "--foo" is a custom argument that takes a parameter.
+tldr --foo bar bash
+```
+
+Furthermore, clients may OPTIONALLY support a colon to specify the platform name in addition to the `--platform`, `-p` syntax described above Here are some examples:
+
+```bash
+tldr platform:page
+tldr windows:type
+tldr common:git checkout
```

Comment:
This is **really really** confusing. People used to cli conventions are likely read it like two arguments instead. If _prefix_ syntax is used kebab case should be mandated:
```
$ tldr common:git-checkout
```

Reply:
I see. Good idea! I'll change that now.